pkg/component/controller/kuberouter.go

```diff
@@ -177,6 +177,7 @@ func (k *KubeRouter) Start(_ context.Context) error {
 	return nil
 }
 
+// from https://github.com/cloudnativelabs/kube-router/blob/master/daemonset/
```

Comment:
NB: The diff of generic-kuberouter.yaml between v1.6.1 and v2.1.0:

```diff
diff --git a/daemonset/generic-kuberouter.yaml b/daemonset/generic-kuberouter.yaml
index 8d812848..0179d35c 100644
--- a/daemonset/generic-kuberouter.yaml
+++ b/daemonset/generic-kuberouter.yaml
@@ -60,6 +60,10 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: spec.nodeName
+        - name: POD_NAME
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.name
         - name: KUBE_ROUTER_CNI_CONF_FILE
           value: /etc/cni/net.d/10-kuberouter.conflist
         livenessProbe:
@@ -98,13 +102,19 @@ spec:
             TMP=/etc/cni/net.d/.tmp-kuberouter-cfg;
             cp /etc/kube-router/cni-conf.json ${TMP};
             mv ${TMP} /etc/cni/net.d/10-kuberouter.conflist;
-          fi
+          fi;
+          if [ -x /usr/local/bin/cni-install ]; then
+            /usr/local/bin/cni-install;
+          fi;
         volumeMounts:
         - mountPath: /etc/cni/net.d
           name: cni-conf-dir
         - mountPath: /etc/kube-router
           name: kube-router-cfg
+        - name: host-opt
+          mountPath: /opt
       hostNetwork: true
+      hostPID: true
       tolerations:
       - effect: NoSchedule
         operator: Exists
@@ -126,6 +136,9 @@ spec:
         hostPath:
           path: /run/xtables.lock
           type: FileOrCreate
+      - name: host-opt
+        hostPath:
+          path: /opt
 
 ---
 apiVersion: v1
@@ -169,7 +182,29 @@ rules:
       - get
       - list
       - watch
-
+  - apiGroups:
+      - "coordination.k8s.io"
+    resources:
+      - leases
+    verbs:
+      - get
+      - create
+      - update
+  - apiGroups:
+      - ""
+    resources:
+      - services/status
+    verbs:
+      - update
+  - apiGroups:
+      - "discovery.k8s.io"
+    resources:
+      - endpointslices
+    verbs:
+      - get
+      - list
+      - watch
+      
 ---
 kind: ClusterRoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
```

The cni-install and /opt mounting is related to the CNI binary installation, which the official kube-router image is capable of. K0s has its own CNI installation image, so it's fine to leave that out.